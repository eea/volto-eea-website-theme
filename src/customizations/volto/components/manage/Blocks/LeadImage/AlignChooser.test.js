import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { render } from '@testing-library/react';
import AlignChooser from './AlignChooser';

const mockStore = configureStore();

describe('AlignChooser', () => {
  const props = {
    align: 'left',
    onChangeBlock: jest.fn(),
    data: { copyrightPosition: 'left' },
    block: '123',
    actions: ['left', 'right', 'center', 'full'],
  };
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });

  it('renders the align buttons', () => {
    const { container } = render(
      <Provider store={store}>
        <AlignChooser {...props} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('calls onChangeBlock when an align button is clicked', () => {
    const { container } = render(
      <Provider store={store}>
        <AlignChooser {...props} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('marks the active align button as active', () => {
    const { container } = render(
      <Provider store={store}>
        <AlignChooser {...props} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
