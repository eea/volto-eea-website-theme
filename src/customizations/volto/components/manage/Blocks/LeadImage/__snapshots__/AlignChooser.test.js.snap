// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignChooser calls onChangeBlock when an align button is clicked 1`] = `
<div>
  <div
    class="align-buttons"
  >
    <div
      class="ui buttons"
    >
      <button
        aria-label="Left"
        class="ui active basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Right"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Center"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Full"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`AlignChooser marks the active align button as active 1`] = `
<div>
  <div
    class="align-buttons"
  >
    <div
      class="ui buttons"
    >
      <button
        aria-label="Left"
        class="ui active basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Right"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Center"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Full"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`AlignChooser renders the align buttons 1`] = `
<div>
  <div
    class="align-buttons"
  >
    <div
      class="ui buttons"
    >
      <button
        aria-label="Left"
        class="ui active basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Right"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Center"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
    <div
      class="ui buttons"
    >
      <button
        aria-label="Full"
        class="ui basic icon button"
      >
        <svg
          class="icon"
          style="height: 24px; width: auto; fill: currentColor;"
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
  </div>
</div>
`;
